docs/source/howto/data.rst

```diff
@@ -248,11 +247,247 @@ The node class has various methods to set these attributes, such as :py:`~aiida.
+=============================
+
+Once you have successfully completed a series of workflows for your project, or have imported a dataset you are interested in, you want to quickly find the data that is relevant for your analysis.
+The data in an AiiDA database is stored as a graph of connected entities, which can be easily *queried* with the ``QueryBuilder`` class.
+
+The ``QueryBuilder`` lets you query your AiiDA database independently of the backend used under the hood.
+Before starting to write a query, it helps to:
+
+*   know what you want to query for.
+    In the language of databases, you need to tell the backend what to *project*.
+    For example, you might be interested in the label of a calculation and the PK's of all its outputs.
+*   know the relationships between entities you are interested in.
+    Nodes of an AiiDA graph (vertices) are connected with links (edges).
+    A node can for example be either the input or output of another node, but also an ancestor or a descendant.
+*   know how you want to filter the results of your query.
+
+Once you are clear about what you want and how you can get it, the QueryBuilder will build an SQL-query for you.
+
+There are two ways of using the QueryBuilder:
+
+#.  In the *appender* method, you construct your query step by step using the ``QueryBuilder.append()`` method.
+#.  In the *queryhelp* approach, you construct a dictionary that defines your query and pass it to the ``QueryBuilder``.
+
+Both APIs provide the same functionality - the appender method may be more suitable for interactive use, e.g., in the ``verdi shell``, whereas the queryhelp method can be useful in scripting.
+In this how-to we will focus on the basics of the appender method.
+For more advanced queries or more details on the queryhelp, see the :ref:`topics section on advanced querying <topics:database:advancedquery>`.
+
+.. _how-to:data:find:select:
+
+Selecting entities
+------------------
+
+Using the ``append()`` method of the ``QueryBuilder``, you can query for the entities you are interested in.
+Suppose you want to query for calculation job nodes in your database:
+
+.. code-block:: python
+
+    from aiida.orm.querybuilder import QueryBuilder
+    qb = QueryBuilder()       # Instantiating instance. One instance -> one query
+    qb.append(CalcJobNode)    # Setting first vertex of path
+
+If you are interested in instances of different classes, you can also pass a tuple, list or set of classes.
+However, they have to be of the same ORM-type (e.g. all have to be subclasses of ``Node``):
+
+.. code-block:: python
+
+    qb = QueryBuilder()       # Instantiating instance. One instance -> one query
+    qb.append([CalcJobNode, WorkChainNode]) # Setting first vertice of path, either WorkChainNode or Job.
+
+.. note::
+
+    Remember that :ref:`processes<topics:processes:concepts:types>` have both a run-time ``Process`` that steers them and a ``Node`` that stores their metadata in the database.
```

Comment:
1. What exactly does *steers* mean in this context? (I of course have an idea, but we should be precise.)
2. A ``Node`` can store more than metadata, right?

Reply:
1. Perhaps use "executes" instead or "steers"?
2. Correct! I've changed "metadata" to "data".

Reply:
Re 1: I think the issue is more that the relationship between the mentioned entities is unclear. But I don't want to hold up the PR because of this. If we can't clarify that, let's just leave it *as-is*.

Reply:
I agree that it is not an easy-to-grasp concept, which is why I explained it in detail in [this topic section](https://aiida.readthedocs.io/projects/aiida-core/en/docs-revamp/topics/processes/concepts.html#concepts). Maybe referring to that for details would be sufficient.